src/compiler/utilities.ts

```diff
@@ -7539,13 +7539,14 @@ export function getCheckFlags(symbol: Symbol): CheckFlags {
 
 /** @internal */
 export function getDeclarationModifierFlagsFromSymbol(s: Symbol, isWrite = false): ModifierFlags {
-    if (s.valueDeclaration) {
+    const checkFlags = getCheckFlags(s);
+    if (!(checkFlags & CheckFlags.ReverseMapped) && s.valueDeclaration) {

```

Comment:
Since I added `.valueDeclaration` to reverse mapped properties, we need to ignore this here - otherwise, we break stripping of `readonly` modifiers that is required by https://github.com/microsoft/TypeScript/pull/12589